exp/api/v1alpha3/awsmanagedmachinepool_types.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha3"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/errors"
+)
+
+const (
+	// ManagedMachinePoolFinalizer allows the controller to clean up resources on delete
+	ManagedMachinePoolFinalizer = "awsmanagedmachinepools.infrastructure.cluster.x-k8s.io"
+)
+
+var (
+	// DefaultEKSNodegroupRole is the name of the default IAM role to use for EKS nodegroups
+	// if no other role is supplied in the spec and if iam role creation is not enabled. The default
+	// can be created using clusterawsadm or created manually
+	DefaultEKSNodegroupRole = fmt.Sprintf("eks-nodegroup%s", infrav1.DefaultNameSuffix)
+)
+
+// AWSManagedMachinePoolSpec defines the desired state of AWSManagedMachinePool
+type AWSManagedMachinePoolSpec struct {
+	// EKSNodegroupName specifies the name of the nodegroup in AWS
+	// corresponding to this MachinePool. If you don't specify a name
+	// then a default name will be created based on the namespace and
+	// name of the managed machine pool.
+	// +optional
+	EKSNodegroupName string `json:"eksNodegroupName,omitempty"`
+	// Subnets specifies which subnets are used for the
+	// auto scaling group of this nodegroup
+	// +optional
+	Subnets []string `json:"subnets,omitempty"`
```

Comment:
Hmm I need to check if the API accepts `ARN`s, if not I think changing the type might be a mistake because a user can specify an `ARN` without an `ID`. (at least the field here should be titled `SubnetIDs`)

Reply:
Good point.....calling it SubnetIDs  is good